smarts/core/remote_agent_buffer.py

```diff
@@ -93,6 +93,16 @@ def __init__(self, zoo_manager_addrs=None, buffer_size=3, max_workers=4):
             self._remote_agent_future() for _ in range(self._buffer_size)
         ]
 
+        # Catch abrupt terminate signals
+        signal.signal(signal.SIGTERM, self._stop_servers)
+
+    def _stop_servers(self, *args):
+        self.destroy()
+        self._log.debug(
+            f"Shutting down zoo manager and zoo workers gracefully due to abrupt process stop."
+        )
+        sys.exit(0)
```

Comment:
I had a small chuckle seeing gracefully on one line and `sys.exit()` immediately following.

Reply:
Removed the phrase `gracefully`.